### DISCUSSION TITLE
Initiating Client instances

### DISCUSSION THREAD

**Discussion by clhettiarachchi** [AlexaCRM/dynamics-webapi-toolkit#111] [Q&A] [answered]
```
add_action('wpcf7_mail_sent', function ($contact_form) {

  $client = \AlexaCRM\WebAPI\ClientFactory::createOnlineClient($host, $appId, $secret);

  // Code to create entity
}
```

Here I'm going to use Contact Form 7 hook to create an Entity. Is it correct to initiate the $client instance on every call? Any other solution?

**Comment by georged** [comment, marked as answer]
`createOnlineClient` takes logger and cache pool as optional parameters. If cache pool is set, we will cache the access token and bypass the call to the authentication endpoint, if possible. It does look like you're using the client inside the [Dataverse plugin](https://wordpress.org/plugins/integration-cds). In that instance you should be using the client provided by the plugin that takes care of caching: 

```php
$webapi = ConnectionService::instance()->getClient();
$webapi->Create(...);
```

**Reply by clhettiarachchi** [reply]
It works well. Thank you!
